### ISSUE TITLE
Android Store OpenSSL Security Warning

### ISSUE THREAD

**Issue by sLatzko** [MobileChromeApps/mobile-chrome-apps#528]
Hello everyone!

Recently I uploaded my cca-project to the android app store whereupon I've got a security warning which states that the openssl version is too low (Heartbleed-Bug).

After researching I figured out it has to do with the cross-walk version, which is 8 (cca 0.5.1).
https://crosswalk-project.org/jira/browse/XWALK-3217

The problem could be resolved be updating cross-walk. 
https://github.com/MobileChromeApps/mobile-chrome-apps/issues/465

I tried to update it manually,
https://github.com/MobileChromeApps/mobile-chrome-apps/blob/master/dev-docs/cloning-this-repo.md

but couldn't resolve some issues because apparently the master branch is broken.
https://github.com/MobileChromeApps/mobile-chrome-apps/issues/524

Trying to link cordova-cli and cordova-lib did not work for me - maybe due to lack of experience with npm.

What I actually want to know is when a new cca release with a higher crosswalk version will be available?

Thank you

**Comment by agrieve** [comment]
Long overdue, I know!
Just published a release candidate that includes newer crosswalk. Would be great if you wanted to try it out!

    npm install -g cca@0.6.0-rc1

**Comment by sLatzko** [comment]
thats great news! I'll try it right now!

**Comment by sLatzko** [comment]
Alright, I've upgraded cca to 0.6.0-rc1.

I guess there has been changed something about the manifest.json because the information doesn't get used anymore. I guessed I could try to change mobile.manifest.json which looks like this right now:
```json
{
	"packageId": "com.JohanniterDienstplan",
	"name": "Johanniter Dienstplan",
	"version": "0.2",
	"manifest_version": 2,
	"app": {
		"background": {
			"scripts": ["background.js"]
		}
	},
/*	"icons": {
		"16": "icons/icon_16.png",
		"36": "icons/icon_36.png",
		"48": "icons/icon_48.png",
		"57": "icons/icon_57.png",
		"72": "icons/icon_72.png",
		"78": "icons/icon_78.png",
		"96": "icons/icon_96.png",
		"114": "icons/icon_114.png",
		"120": "icons/icon_120.png",
		"128": "icons/icon_128.png",
		"144": "icons/icon_144.png",
		"152": "icons/icon_152.png"
	},*/
	"permissions": [
		"https://dp.johanniter.at/*",
		"storage"
	],
	"minimum_chrome_version": "28",
	// Android-only overrides here (e.g. icons, name, version)
	"android": {},
	// iOS-only overrides here (e.g. icons, name, version)
	"ios": {},
	// Uncomment to use system WebView rather than Crosswalk
	// "webview": "system",
	// Uncomment to enable use of eval()
	// "cspUnsafeEval": true,
	// Uncomment to enable use of inline <script>
	// "cspUnsafeInline": true,
}
```
Before this change a white screen appeared showing "Hello, mobile".

Afterwards I've gotten an error - Failed Version Downgrade. 
But I don't understand why because the emulator uses android 5.0

```
Failed to launch app on emulator: Failed to install apk to emulator: 	pkg: /data/local/tmp/android-x86-debug.apk
Failure [INSTALL_FAILED_VERSION_DOWNGRADE]

Error: /Users/Stefan/joh-dpa/platforms/android/cordova/run: Command failed with exit code 1
    at ChildProcess.whenDone (/usr/local/lib/node_modules/cca/node_modules/cordova/node_modules/cordova-lib/src/cordova/superspawn.js:131:23)
    at ChildProcess.emit (events.js:110:17)
    at maybeClose (child_process.js:1008:16)
    at Process.ChildProcess._handle.onexit (child_process.js:1080:5)
```

Any guesses?

**Comment by schme16** [comment]
Failed version downgrade is due to the way the internal build number seem to work with Chrome Mobile Apps when you switch into and out of using crosswalk (or in this case when you upgrade your crosswalk); just bump the version number up and you should be good to go

**Comment by agrieve** [comment]
Or uninstall the app from the device before running.

**Comment by sLatzko** [comment]
Thank you for your help.

Just to inform you. I recreated my cca project and found out that

```
cca create projectName --link-to:Path/to/Project
````
doesn't include the project files anymore. I copied all files manually and got it up and running for now. I also had to change the *manifest.json*

**Comment by agrieve** [comment]
Thanks for trying it out! I don't think `--link-to:path` has ever worked. Try `--link-to=path` or `--link-to path`.

What did you have to change in manifest.json? More than the version?

**Comment by sLatzko** [comment]
You're right, my fault. Sorry. `--link-to=path` worked as expected. Thus no need to change the manifest.json.

But something different.

While building and deploying the cca project per terminal works normally, I've problems sorting out error messages in eclipse after I've used `Project > clean`. It seems to me those two packages are missing.

```java
import org.xwalk.core.XWalkNavigationHistory;
import org.xwalk.core.XWalkView;
```

Have I missed something?

**Comment by agrieve** [comment]
We've switched to using Crosswalk's maven repository rather than bundling a copy of their .jar / .so. This works fine with Gradle, but I don't think it works on Eclipse (at least not by default). Android Studio works fine if you can stomach an IDE switch.

**Comment by sLatzko** [comment]
Alright, won't hurt me. 

Just for the record. The security warning has been dismissed some hours after I uploaded the new version.

**Comment by agrieve** [comment]
Great to hear! Just released `cca@0.6.0`, so closing :)
